### PR TITLE
Closes #2230 update_generic: Add new functions and handling of NA for sorting variables

### LINKED ISSUE #2230: Update Generic.Rmd
Update Generic.Rmd such that it covers all the updates from the last releases and provides a complete overview of the generic functions.

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
@bundfussr can we pin the obs function discussion to this PR as well??

File: NEWS.md [hunk 1]
```diff
@@ -179,6 +179,9 @@ the imputation functions, e.g., `derive_vars_dtm()`. (#2222)
 - All "Example Script(s)" sections in the User Guide vignettes were updated to point the user towards using `use_ad_template("ADaM")` rather 
   than linking to the template in the code repository. (#2239)
 
+- Handling of `NA` values was added to the documentation of the `order` argument
+for all functions. (#2230)
```
Comment (bms63) [inline NEWS.md:183]: Can #2257 also be referenced since this issue spurred the discussion?

File: R/derive_extreme_event.R [hunk 2]
```diff
@@ -33,6 +33,8 @@
 #'   If a particular event from `events` has more than one observation, within
 #'   the event and by group, the records are ordered by the specified order.
 #'
+#'   `r roxygen_order_na_handling()`
```
Comment (bms63) [inline R/derive_extreme_event.R:36]: This is so cool!!! 
image

Comment (bms63) [reply R/derive_extreme_event.R:36]: @pharmaverse/admiral This is a very cool use of the our `roxygen2.R` file with linking to a vignette for some extra guidance!!  Keep this one in your backpocket!

**Review by bms63 — COMMENTED** [review]
I was poking around the vignettes seeing if we have any missing data examples that this might be nice to cross-link to.

What about here: https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values

and here: https://pharmaverse.github.io/admiral/reference/convert_blanks_to_na.html?q=NAs#null

This was just a 5 second scan

**Review by kaz462 — APPROVED** [review]

**Review by ynsec37 — COMMENTED** [review]
`if_else(is.na(AVISITN), -Inf, AVISITN)` only work for numeric,
We can try to sort by `!is.na(VISIT), VISIT`, that work for both numeric and character

```
advs <- tribble(
  ~USUBJID, ~PARAMCD, ~AVISITN, ~AVAL, ~VISIT,
  "1",      "WEIGHT",       NA,  62.1, NA,
  "1",      "WEIGHT",        1,  62.3, "V1",
  "1",      "WEIGHT",        2,  62.5, "V2",
  "1",      "WEIGHT",        3,  62.4, "V3"
)

derive_var_extreme_flag(
  advs,
  by_vars = exprs(USUBJID, PARAMCD),
  order = exprs(VISIT),
  mode = "last",
  new_var = LSTVISFL
)

derive_var_extreme_flag(
  advs,
  by_vars = exprs(USUBJID, PARAMCD),
  order = exprs(if_else(is.na(VISIT), -Inf, VISIT)),
  mode = "last",
  new_var = LSTVISFL
)

derive_var_extreme_flag(
  advs,
  by_vars = exprs(USUBJID, PARAMCD),
  order = exprs(!is.na(VISIT), VISIT),
  mode = "last",
  new_var = LSTVISFL
)
```
image

File: vignettes/generic.Rmd [hunk 3]
```diff
@@ -399,6 +401,43 @@ However, the "joined" functions require much more resources (time and memory),
 hence it is recommended to use them only if it is really required, i.e., the
 condition for selecting records depends on variables from _both_ datasets.
 
+### Sort Order {#sort_order}
+
+The `{admiral}` functions use `dplyr::arrange()` for sorting, i.e., `NA`s are
+always sorted to the end (regardless whether `desc()` is used or not).
+
+Consider for example the following derivation of a last visit flag. The record
+with `AVISITN == NA` is flagged because `NA` is sorted to the end.
```
Comment (manciniedoardo) [inline vignettes/generic.Rmd:410]: Do you want `=` or `==` here?

File: NEWS.md (hunk 1 above)
Comment (bundfussr) [reply NEWS.md]: Yes, good point.

**Comment by bundfussr** [comment]
> I was poking around the vignettes seeing if we have any missing data examples that this might be nice to cross-link to.
> 
> What about here: https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values
> 
> and here: https://pharmaverse.github.io/admiral/reference/convert_blanks_to_na.html?q=NAs#null
> 
> This was just a 5 second scan

I have added a link from the "Get Started" page to the "Sort Order" section.

`convert_blanks_to_na()` is already mentioned in the "Handling of Missing Values" section. So I am not sure how it should be cross-linked.

**Comment by bundfussr** [comment]
> `if_else(is.na(AVISITN), -Inf, AVISITN)` only work for numeric,
> We can try to sort by `!is.na(VISIT), VISIT`, that work for both numeric and character

@ynsec37 , good point! Thanks for providing this example. I have added it to the vignette.

**Review by manciniedoardo — APPROVED** [review]

**Comment by bms63** [comment]
Let's go!

**Pull request merged**